runtime/src/lib.rs

```diff
@@ -126,59 +86,73 @@ fn handle_command(drivers: &mut Drivers) -> CaliptraResult<MboxStatusE> {
+    // Some responses have a variable size
+    // Leave as None otherwise
+    let resp_size_ovrd: Option<usize> = None;
+
+    // Handle the request and generate the response
+    let mut resp: Option<MailboxResp> = match CommandId::from(req_packet.cmd) {
+        CommandId::FIRMWARE_LOAD => Err(CaliptraError::RUNTIME_UNIMPLEMENTED_COMMAND),
         CommandId::GET_IDEV_CSR => Err(CaliptraError::RUNTIME_UNIMPLEMENTED_COMMAND),
         CommandId::GET_LDEV_CERT => Err(CaliptraError::RUNTIME_UNIMPLEMENTED_COMMAND),
         CommandId::ECDSA384_VERIFY => {
             verify::handle_ecdsa_verify(drivers, cmd_bytes)?;
-            Ok(MboxStatusE::CmdComplete)
+            Ok(None)
         }
         CommandId::STASH_MEASUREMENT => Err(CaliptraError::RUNTIME_UNIMPLEMENTED_COMMAND),
         CommandId::INVOKE_DPE => Err(CaliptraError::RUNTIME_UNIMPLEMENTED_COMMAND),
         #[cfg(feature = "test_only_commands")]
         CommandId::TEST_ONLY_GET_LDEV_CERT => {
-            let mut cert = [0u8; 1024];
-            let cert_len = dice::copy_ldevid_cert(&drivers.data_vault, &mut cert)?;
-            drivers.mbox.write_response(
-                cert.get(..cert_len)
-                    .ok_or(CaliptraError::RUNTIME_INSUFFICIENT_MEMORY)?,
-            )?;
-            Ok(MboxStatusE::DataReady)
+            let (payload, resp_size_ovrd) = dice::handle_get_ldevid_cert(&drivers.data_vault)?;
+            Ok(Some(MailboxResp::TestGetCert(payload)))
         }
         #[cfg(feature = "test_only_commands")]
         CommandId::TEST_ONLY_GET_FMC_ALIAS_CERT => {
-            let mut cert = [0u8; 1024];
-            let cert_len = dice::copy_fmc_alias_cert(&drivers.data_vault, &mut cert)?;
-            drivers.mbox.write_response(
-                cert.get(..cert_len)
-                    .ok_or(CaliptraError::RUNTIME_INSUFFICIENT_MEMORY)?,
-            )?;
-            Ok(MboxStatusE::DataReady)
+            let (payload, resp_size_ovrd) = dice::handle_get_fmc_alias_cert(&drivers.data_vault)?;
+            Ok(Some(MailboxResp::TestGetCert(payload)))
+        }
+        CommandId::VERSION => {
+            FipsModule::version(drivers)?;
+            Ok(None)
+        }
+        CommandId::SELF_TEST => {
+            FipsModule::self_test(drivers)?;
+            Ok(None)
+        }
+        CommandId::SHUTDOWN => {
+            FipsModule::shutdown(drivers)?;
+            Ok(None)
         }
-        CommandId::VERSION => FipsModule::version(drivers),
-        CommandId::SELF_TEST => FipsModule::self_test(drivers),
-        CommandId::SHUTDOWN => FipsModule::shutdown(drivers),
         _ => Err(CaliptraError::RUNTIME_UNIMPLEMENTED_COMMAND),
-    }
+    }?;
+
+    // If there was no reponse payload from the handler, still send back a header
+    let mut resp = match resp {
+        Some(resp) => resp,
+        None => MailboxResp::Header(MailboxRespHeader::default()),
+    };
```

Comment:
Could just return `MailboxResp::Header(MailboxRespHeader::default())` from commands that only return the header. That way the code is easier to follow.